src/OpenTelemetry.AutoInstrumentation/Configuration/MeterSettings.cs

```diff
@@ -0,0 +1,159 @@
+// <copyright file="MeterSettings.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenTelemetry.AutoInstrumentation.Configuration
+{
+    /// <summary>
+    /// Meter Settings
+    /// </summary>
+    public class MeterSettings : Settings
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MeterSettings"/> class
+        /// using the specified <see cref="IConfigurationSource"/> to initialize values.
+        /// </summary>
+        /// <param name="source">The <see cref="IConfigurationSource"/> to use when retrieving configuration values.</param>
+        private MeterSettings(IConfigurationSource source)
+            : base(source)
+        {
+            MetricExporter = ParseMetricExporter(source);
+            ConsoleExporterEnabled = source.GetBool(ConfigurationKeys.Metrics.ConsoleExporterEnabled) ?? false;
+
+            var instrumentations = new Dictionary<string, MeterInstrumentation>();
+            var enabledInstrumentations = source.GetString(ConfigurationKeys.Metrics.Instrumentations);
+            if (enabledInstrumentations != null)
+            {
+                foreach (var instrumentation in enabledInstrumentations.Split(separator: ','))
+                {
+                    if (Enum.TryParse(instrumentation, out MeterInstrumentation parsedType))
+                    {
+                        instrumentations[instrumentation] = parsedType;
+                    }
+                    else
+                    {
+                        throw new ArgumentException($"The \"{instrumentation}\" is not recognized as supported metrics instrumentation and cannot be enabled");
+                    }
+                }
+            }
+
+            var disabledInstrumentations = source.GetString(ConfigurationKeys.Metrics.DisabledInstrumentations);
```

Comment:
I know that we have exactly same config structure for traces, but I have some doubts that we need it.
For now, it can be replaced by just removing the instrumentation from the enabled instrumentations list. It will change if we enable the metrics by default.

Reply:
I was thinking about the same as well.
But then I thought that it can be used for `Integrations` as well (in the same way as for traces). Nothing for adding byte-code metrics instrumentation, right?

Reply:
Correct.

Below is the code from `EnvironmentConfigurationTracerHelper.cs`. Instrumentation list is parsed directly and added to `TracerProviderBuilder`. Incase of metrics addition is required to `MeterProviderBuilder`. Instrumentation libraries supported for traces and metrics as different and this needs to be handled separately. OpenTelemetry SDK treats traces, metrics and logs as 3 different signals and provides and option to enable/disable all 3 signals on a need basis. Design motivation is same here to treat all the signals different.

```
    public static TracerProviderBuilder UseEnvironmentVariables(this TracerProviderBuilder builder, TracerSettings settings)
    {
        var resourceBuilder = ResourceBuilder.CreateDefault();

        builder
            .SetResourceBuilder(resourceBuilder)
            .SetExporter(settings);

        foreach (var enabledInstrumentation in settings.EnabledInstrumentations)
        {
            if (AddInstrumentation.TryGetValue(enabledInstrumentation, out var addInstrumentation))
            {
                addInstrumentation(builder);
            }
        }

        builder.AddSource(settings.ActivitySources.ToArray());
        foreach (var legacySource in settings.LegacySources)
        {
            builder.AddLegacySource(legacySource);
        }

        return builder;
    }

```

Reply:
Are there any requested changes here? If not, I think we can merge the PR in its current state

Reply:
Similar question has come during Environment variables too. It's better to discuss the topic in next SIG meeting. I will get the PR merged to unblock the metrics integration.